kmk/extensions/ble_split.py

```diff
@@ -0,0 +1,154 @@
+from adafruit_ble import BLERadio
+from adafruit_ble.advertising.standard import ProvideServicesAdvertisement
+from adafruit_ble.services.nordic import UARTService
+from kmk.extensions import Extension
+from kmk.hid import HIDModes
+from kmk.kmktime import ticks_diff, ticks_ms
+from kmk.matrix import intify_coordinate
+
+
+class BLE_Split(Extension):
+    def __init__(
+        self, split_flip=True, split_side=None, split_offsets=[], hid_type=HIDModes.BLE
+    ):
+        self._is_target = True
+        self._uart_buffer = []
+        self.hid_type = hid_type
+        self.split_flip = split_flip
+        self.split_side = split_side
+        self.split_offsets = split_offsets
+        self._ble = BLERadio()
+        self._ble_last_scan = ticks_ms() - 5000
+        self._is_target = True
+        self._connection_count = 0
+        self._uart = None
+        self._uart_connection = None
+        self._advertisment = None
+        self._advertising = False
+
+    def __repr__(self):
+        return f'BLE_SPLIT({self._to_dict()})'
+
+    def _to_dict(self):
+        return f'BLE_Split( _ble={self._ble} _ble_last_scan={self._ble_last_scan} _is_target={self._is_target} _uart_buffer={self._uart_buffer} _split_flip={self.split_flip} _split_side={self.split_side} _split_offsets={self.split_offsets} )'
+
+    def during_bootup(self, keyboard):
+        if self.split_side == 'Left':
```

Comment:
this could probs be simplified to `self._is_target = self.split_side == 'Left'`

Reply:
Not super familiar with that syntax, but yeah, it definitely could. That was a dirty port from when split_side could change due to wired being different. Good catch. I think that `self._is_target = bool(self.split_side == 'Left')` is also valid.